client/network/src/service.rs

```diff
@@ -157,12 +157,14 @@ impl<B: BlockT + 'static, H: ExHashT> NetworkWorker<B, H> {
 				Role::Sentry { validators } => {
 					for validator in validators {
 						sentries_and_validators.insert(validator.peer_id.clone());
+						reserved_nodes.insert(validator.peer_id.clone());
 						known_addresses.push((validator.peer_id.clone(), validator.multiaddr.clone()));
 					}
 				}
 				Role::Authority { sentry_nodes } => {
 					for sentry_node in sentry_nodes {
 						sentries_and_validators.insert(sentry_node.peer_id.clone());
+						reserved_nodes.insert(sentry_node.peer_id.clone());
```

Comment:
In most setups I would expect sentry nodes to have a public and a private IP addresses. From the outside nodes can reach a sentry node via its public IP addresses, validators can reach their sentry node via their private IP address.

I would expect `sentry_nodes` to contain the **public** multiaddresses of the sentry nodes of a validator.

Does it make sense to have a validator connect to its sentry node via its public address as a reserved node?